Replacing deprecated constants, adding and fixing stories

dhruvv173 opened the pull request:
## Explanation

- This PR is a part of the following issues
  - #19526 
  - #18896 
  - #18714 
  - #19555
 
Replaced deprecated consts in the following folders

- `ui/components/app/snaps/update-snap-permission-list`
- `ui/components/app/snaps/snap-privacy-warning`
- `ui/components/app/snaps/snap-settings-card`
- Created Story for `SnapInstallWarning`
- Fixed story error for `SnapSettingsCard`


<!--
Thanks for the pull request. Take a moment to answer these questions so that reviewers have the information they need to properly understand your changes:

* What is the current state of things and why does it need to change?
* What is the solution your changes offer and how does it work?

Are there any issues, Slack conversations, Zendesk issues, user stories, etc. reviewers should consult to understand this pull request better? For instance:

-->

## Screenshots/Screencaps

<!-- If you're making a change to the UI, make sure to capture a screenshot or a short video showing off your work! -->
### Fixing `SnapSettingCard` story
### Before
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/caa6de54-8870-4e12-a5e8-4f1176135f88)

### After
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/61f3213b-776a-40db-885f-32ae6ebaeec5)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/62c3dd9b-baec-43e8-83d8-24d60533b5fb)

#### `targetedMetadata` propType [error](https://github.com/MetaMask/metamask-extension/pull/19584#discussion_r1230180132) fixed
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/4172e78a-c3e5-4bb3-812e-b04f96d93e64)



### Creating Story for `SnapInstallWarning`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/53652dcf-ee11-469b-9194-b9e5f252c9c7)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/e50f5f47-ab62-433f-bfbb-b45a552affc2)

Fixing the [error](https://github.com/MetaMask/metamask-extension/pull/19584#issuecomment-1592017104) while trying to click outside the Box

https://github.com/MetaMask/metamask-extension/assets/79097544/8d420824-1fbe-4035-aefe-667c07bafe08





### Before `UpdateSnapPermissionList`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/97145365-99b4-459b-94aa-319cebd33a36)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/57415fbf-9d28-4134-b620-88aac8aa84f8)
### After `UpdateSnapPermissionList`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/34cdea77-6741-4632-a8a6-a922c9b3784d)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/21069e41-9740-4394-af46-a5804639ee65)

### Before `SnapPrivacyWarning`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/e142b039-163c-4cac-b928-bce42896ff9f)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/fa6043bb-7135-445c-9659-9f5cc3ad7a4c)
### After `SnapPrivacyWarning`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/51b43463-545a-47be-ac34-747a523f2129)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/1998a40b-f8d1-46c2-a290-9946f940cb78)

### Before `SnapUpdate`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/968fbdcd-f618-4b51-9277-93906b314d9e)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/82331816-f34d-44e4-80f2-1db5e4f7a6aa)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/24cd1ffd-94f5-4c6f-96d3-0422c97c8936)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/0b73ae28-5f38-429c-8090-7a0309f63b2f)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/864e13f4-c3db-40a1-aecc-01c58038dad6)

### After `SnapUpdate`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/2ab45ebc-0f3b-42fb-ab06-53cae4612440)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/45d1b43d-1200-472a-934f-8973699c1e7f)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/300cde5d-afb0-4610-b72c-616425e12d99)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/c0e07a93-7dbf-4495-86d9-0685a470f9c9)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/f5ffd92c-eeac-4907-b158-41d37c230ad1)

### Before `SnapConnect`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/f9e67d76-c8ba-4151-a1fb-57588c6c667d)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/63a46344-9e8f-4b78-8a85-b468892522e1)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/98222584-1065-4a10-8daf-9b41a4b0e62b)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/7d12d853-14fd-4791-b65e-453929fcebb1)

### After `SnapConnect`
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/71e3d726-d65f-4f17-aab4-826d8d39204f)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/ff54f8f2-b44b-43d0-b8bb-45ee8e3e1fa6)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/528e20c8-a001-48d1-a48e-4fa5a2690a14)

## Manual Testing Steps

<!--
How should reviewers and QA manually test your changes? For instance:

- Go to this screen
- Do this
- Then do this
-->

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

dhruvv173 committed:
Replacing deprecated consts; adding,fixing stories

dhruvv173 committed:
Merge branch 'develop' into issue/replacing-deprecated/snaps

dhruvv173 committed:
moving snap-permission-list to Snaps folder

dhruvv173 committed:
Merge branch 'issue/replacing-deprecated/snaps' of github.com:dhruvv173/metamask-extension into issue/replacing-deprecated/snaps

dhruvv173 commented:
hey @georgewrmarshall, apologies for the number of files included in the PR. Initially, I had only made changes to 4 files but when I was going through the page for Before screenshots I found an error in `SnapSettingCard` story so I tried fixing it while I was at it. I hope you dont mind. I'll make sure to open separate PRs next time. Apologies for the inconvenience.
Thanks!

georgewrmarshall committed:
Replacing more deprecated components and updating stories

georgewrmarshall commented on ui/components/app/snaps/snap-install-warning/index.scss:
Removing unneeded CSS

georgewrmarshall reviewed (commented):
Hey @dhruvv173, great updates! I saw some more deprecated components that could be replaced

https://github.com/MetaMask/metamask-extension/issues/19555

Would you checking my updates on your end and updating the screenshots in you PR?

dhruvv173 commented:
> Hey @dhruvv173, great updates! I saw some more deprecated components that could be replaced
> 
> https://github.com/MetaMask/metamask-extension/issues/19555
> 
> Would you checking my updates on your end and updating the screenshots in you PR?
> 

No worries, I'll replace the remaining deprecated components and update the screenshots.

georgewrmarshall committed:
Lint fixes

dhruvv173 committed:
lint fix

dhruvv173 commented on ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.stories.js:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
-  title: 'Components/App/UpdateSnapPermissionList',
+  title: 'Components/App/Snaps/UpdateSnapPermissionList',
   component: UpdateSnapPermissionList,
 
```
as the `targetedSubjectMetadata` is not passed in the story it throws an error in console
can the object be passed as the one as pasted below?
```
{
    name: PropTypes.string,
    origin: PropTypes.string.isRequired,
    subjectType: PropTypes.string.isRequired,
    extensionId: PropTypes.string,
    iconUrl: PropTypes.string,
}
```

dhruvv173 commented:
hey @georgewrmarshall I have updated the screenshots in the PR description and fixed the small lint issue.
Found one error in `SnapInstallWarning` story while trying to click outside the warning box. Is this behaviour expected or its an issue? would appreciate your insight.
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/31749d67-ffb0-4cb1-9b6e-b5044384a015)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/31c5fe23-69da-4222-9cda-8da84ddaa394)

dhruvv173 committed:
Merge branch 'develop' into issue/replacing-deprecated/snaps

georgewrmarshall committed:
Fixing control to use action to prevent prop error and some style updates

georgewrmarshall commented on ui/components/app/snaps/snap-install-warning/snap-install-warning.stories.js:
```diff
@@ -0,0 +1,36 @@
+  title: 'Components/App/snaps/SnapInstallWarning',
+  component: SnapInstallWarning,
+  argTypes: {
+    onCancel: {
+      action: 'onCancel',
+    },
+    onSubmit: {
+      action: 'onSubmit',
+    },
```
Function type props can use the action control. Updated here

dhruvv173 replied:
okay, I'll keep this in mind while adding new stories 👍

georgewrmarshall committed:
Update story to fix targetSubjectMetadata propType error

dhruvv173 commented:
hey @georgewrmarshall I have updated the PR description following your recent commits, please go through it once when possible. Let me know if anything else needs to be addressed otherwise this is ready for review imo.

dhruvv173 committed:
Merge branch 'develop' into issue/replacing-deprecated/snaps

georgewrmarshall commented:
Hey @dhruvv173, it looks like the e2e tests are failing on this PR. Are you familiar with debugging e2e tests? It might be a good opportunity to become familiar with them. I think some of my updates removed an id selector. Would you be able add the id selector to the appropriate component so tests pass? You can read about e2e testing locally in the README

![Uploading Screenshot 2023-06-16 at 8.38.23 AM.png…]()

dhruvv173 commented:
hey @georgewrmarshall I encountered some issues while running the e2e tests locally, preventing me from identifying the root cause of the failing snaps tests. However, I tried to work around it by opening a draft PR and committing to it so that e2e tests can run on changed code. I understand that this is not the best practice and I apologize for using this approach :/
I think the failing snaps test might be related to `snap-install-warning` component, I might be wrong tho, would appreciate your insight on it.
Apologies for not being able to fix the issue :(

georgewrmarshall commented:
Hey @dhruvv173, no worries I can walk you through it

Run
```
build:test:flask
```

Then run the test that is failing in Firefox

```
yarn test:e2e:single test/e2e/snaps/test-snap-bip-32.spec.js --browser=firefox --debug --leave-running
```

dhruvv173 commented:
> Hey @dhruvv173, no worries I can walk you through it
> 
> Run
> 
> ```
> yarn build:test:flask
> ```
> 
> Then run the test that is failing in Firefox
> 
> ```
> yarn test:e2e:single test/e2e/snaps/test-snap-bip-32.spec.js --browser=firefox --debug --leave-running
> ```

thanks for the help! However, when I try to use the built extension I get this error in console and the extension doesnt start.  
```
Error: LavaMoat - required package not in allowlist: package "@metamask/eth-trezor-keyring>@trezor/connect-web>@trezor/connect" requested "browserify>events" as "events"
```
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/30f2180c-7896-4f26-af2f-c7b8cfe69135)

I found an open issue related to it [here](https://github.com/MetaMask/metamask-extension/issues/13253).
There was another similar issue which was related to building without lavamoat. The dev mode works fine for me but theres an issue with test build.

georgewrmarshall committed:
Ensuring state is reset

georgewrmarshall committed:
Merge branch 'develop' into issue/replacing-deprecated/snaps

georgewrmarshall commented:
Hey @dhruvv173, would it be possible to rebase this PR and get it up to date?

dhruvv173 committed:
Merge branch 'develop' into issue/replacing-deprecated/snaps

dhruvv173 committed:
Merge branch 'develop' into issue/replacing-deprecated/snaps

dhruvv173 commented:
> Hey @dhruvv173, would it be possible to rebase this PR and get it up to date?

done!

dhruvv173 committed:
Merge branch 'develop' into issue/replacing-deprecated/snaps

georgewrmarshall commented:
Hey @dhruvv173, I think these updates are great but to ensure we get them it would it be possible to split this PR up into smaller ones that way it will be easier for the snaps team to review. Thanks!

dhruvv173 commented:
> Hey @dhruvv173, I think these updates are great but to ensure we get them it would it be possible to split this PR up into smaller ones that way it will be easier for the snaps team to review. Thanks!

okay will do that over the weekend, thank you for the update!

dhruvv173 commented:
> Hey @dhruvv173, I think these updates are great but to ensure we get them it would it be possible to split this PR up into smaller ones that way it will be easier for the snaps team to review. Thanks!

Done in the below PRs:
- #20301 
- #20296 
- #20294 
- #20284
